cmake/helpers/CheckDependentLibraries.cmake

```diff
@@ -567,7 +575,10 @@ define_find_package2(KEA libkea/KEACommon.h kea)
 gdal_check_package(KEA "Enable KEA driver" CAN_DISABLE)
 
 gdal_check_package(ECW "Enable ECW driver" CAN_DISABLE)
-gdal_check_package(NetCDF "Enable netCDF driver" CAN_DISABLE)
+gdal_check_package(NetCDF "Enable netCDF driver" CAN_DISABLE
+  NAMES netCDF
+  TARGETS netCDF::netcdf NETCDF::netCDF
```

Comment:
this breaks netCDF NC4 detection (NETCDF_HAS_NC4). Perhaps the logic at https://github.com/OSGeo/gdal/blob/master/cmake/modules/packages/FindNetCDF.cmake#L118 should be moved here ?

Reply:
Is NETCDF_HAS_NC4 needed in global scope? 
I would prefer to move it to `frmts/netcdf/CMakeLists.txt`.

Reply:
Or just use variables from netCDFConfig.cmake:
~~~cmake
# Features:
#
set(netCDF_HAS_NC2 yes)
set(netCDF_HAS_NC4 yes)
set(netCDF_HAS_HDF4 no)
set(netCDF_HAS_HDF5 yes)
set(netCDF_HAS_PNETCDF no)
set(netCDF_HAS_PARALLEL no)
set(netCDF_HAS_DAP yes)
set(netCDF_HAS_DAP2 yes)
set(netCDF_HAS_DAP4 yes)
set(netCDF_HAS_DISKLESS yes)
set(netCDF_HAS_MMAP no)
set(netCDF_HAS_JNA no)
~~~

Reply:
> I would prefer to move it to `frmts/netcdf/CMakeLists.txt`.

it should be fine there.

> Or just use variables from netCDFConfig.cmake:

yes we could harmonize the naming of our variable NETCDF_HAS_NC4 with the one of netCDFConfig.cmake.